packages/react-components/src/components/Drawer/DrawerMenu.tsx

```diff
@@ -104,28 +104,51 @@ export const _DrawerMenu = (props: _DrawerMenuProps): JSX.Element => {
       onLightDismiss={props.onLightDismiss}
       heading={props.heading}
     >
-      <Stack styles={props.styles} role="menu" data-ui-id="drawer-menu">
-        {menuItemsToRender?.slice(0, 1).map((item) => (
-          <DrawerMenuItem
-            {...item}
-            key={`${item.itemKey}` + '0'}
-            shouldFocusOnMount={true}
-            styles={modifiedFirstItemStyle}
-            onItemClick={(ev, itemKey) => {
-              onItemClick(item, ev, itemKey);
-            }}
-          />
-        ))}
-        {menuItemsToRender?.slice(1).map((item, i) => (
-          <DrawerMenuItem
-            {...item}
-            key={`${item.itemKey}` + `${i + 1}`}
-            onItemClick={(ev, itemKey) => {
-              onItemClick(item, ev, itemKey);
-            }}
-          />
-        ))}
-      </Stack>
+      {menuItemsToRender !== undefined && menuItemsToRender[0].itemKey === 'reactions' && (
+        <Stack styles={props.styles} role="menu" data-ui-id="drawer-menu">
+          {menuItemsToRender?.slice(0, 1).map((item) => (
+            <DrawerMenuItem
+              {...item}
+              key={`${item.itemKey}` + '0'}
+              shouldFocusOnMount={false}
+              styles={modifiedFirstItemStyle}
+            />
+          ))}
+          {menuItemsToRender?.slice(1).map((item, i) => (
+            <DrawerMenuItem
+              {...item}
+              key={`${item.itemKey}` + `${i + 1}`}
+              onItemClick={(ev, itemKey) => {
+                onItemClick(item, ev, itemKey);
+              }}
+            />
+          ))}
+        </Stack>
+      )}
+      {menuItemsToRender !== undefined && menuItemsToRender[0].itemKey !== 'reactions' && (
+        <Stack styles={props.styles} role="menu" data-ui-id="drawer-menu">
+          {menuItemsToRender?.slice(0, 1).map((item) => (
+            <DrawerMenuItem
+              {...item}
+              key={`${item.itemKey}` + '0'}
+              shouldFocusOnMount={true}
+              styles={modifiedFirstItemStyle}
+              onItemClick={(ev, itemKey) => {
+                onItemClick(item, ev, itemKey);
+              }}
+            />
+          ))}
+          {menuItemsToRender?.slice(1).map((item, i) => (
```

Comment:
This chunk is indentical on each side of the if, only wrap what actually changes in the if-else (or ternary operator in this case).

I _think_ the only difference is not passing in an onItemClick? in which case just make the itemClick optional

Reply:
> I think the only difference is not passing in an onItemClick? in which case just make the itemClick optional

Although based on a difference comment I just made about separting the DrawerMenuItem from reactions, the change become something like:

```diff
-         {menuItemsToRender?.slice(0, 1).map((item) => (
-            <DrawerMenuItem
-              {...item}
-              key={`${item.itemKey}` + '0'}
-              shouldFocusOnMount={false}
-              styles={modifiedFirstItemStyle}
-            />
-          ))}
+        {menuItemsToRender?.slice(0, 1).map((item) => (
+            itemKey === 'reactions' ?
+               <ReactionDrawerMenuItem {props} /> :
+               <DrawerMenuItem {props} />;
```